tests/sentry/api/test_event_search.py

```diff
@@ -561,6 +562,44 @@ def test_blocked_keys(self):
             SearchFilter(key=SearchKey(name="message"), operator="=", value=SearchValue("text")),
         ]
 
+    def test_email_free_text(self):
+        config = SearchConfig.create_from(default_config, allow_free_text_email=True)
+
+        assert parse_search_query(
+            "some_key:123 test.com *@test.com a_1@test.com.uk @name", config=config
+        ) == [
+            SearchFilter(key=SearchKey(name="some_key"), operator="=", value=SearchValue("123")),
+            SearchFilter(
+                key=SearchKey(name="message"), operator="=", value=SearchValue("test.com")
+            ),
+            SearchFilter(
+                key=SearchKey(name="user.email"), operator="=", value=SearchValue("*@test.com")
+            ),
+            SearchFilter(
+                key=SearchKey(name="user.email"), operator="=", value=SearchValue("a_1@test.com.uk")
+            ),
+            SearchFilter(key=SearchKey(name="message"), operator="=", value=SearchValue("@name")),
+        ]
+
+        assert parse_search_query("some_key:123 text", config=config) == [
+            SearchFilter(key=SearchKey(name="some_key"), operator="=", value=SearchValue("123")),
+            SearchFilter(key=SearchKey(name="message"), operator="=", value=SearchValue("text")),
+        ]
+
+        assert parse_search_query("some_key:123 test.com *@test.com a_1@test.com.uk @name") == [
+            SearchFilter(key=SearchKey(name="some_key"), operator="=", value=SearchValue("123")),
+            SearchFilter(
+                key=SearchKey(name="message"), operator="=", value=SearchValue("test.com")
+            ),
+            SearchFilter(
+                key=SearchKey(name="message"), operator="=", value=SearchValue("*@test.com")
+            ),
+            SearchFilter(
+                key=SearchKey(name="message"), operator="=", value=SearchValue("a_1@test.com.uk")
+            ),
+            SearchFilter(key=SearchKey(name="message"), operator="=", value=SearchValue("@name")),
+        ]
```

Comment:
Oh actually, does this match the previous behavior? I would expect it to be a single `SearchFilter`

Reply:
It doesn't :( If the grammar detects something that looks like an email, it automatically visits the email free text function, and I can only add the config within the email free text function so I'm not sure how to make it all a single SearchFilter

Reply:
What happens if we return `None` if config is off?

Reply:
If we return None, the value doesn't get searched, so it'll be like the value wasn't there. So if an email is searched and the config is off, it will be like searching * or searching nothing, and all the replays/issues will be returned in the search